x/valset-pref/msg_server.go

```diff
@@ -25,16 +25,12 @@ var _ types.MsgServer = msgServer{}
 func (server msgServer) SetValidatorSetPreference(goCtx context.Context, msg *types.MsgSetValidatorSetPreference) (*types.MsgSetValidatorSetPreferenceResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	valSetPref, err := server.keeper.SetValidatorSetPreference(ctx, msg.Delegator, msg.Preferences)
+	preferences, err := server.keeper.SetValidatorSetPreference(ctx, msg.Delegator, msg.Preferences)
 	if err != nil {
 		return nil, err
 	}
 
-	setMsg := types.ValidatorSetPreferences{
-		Preferences: valSetPref,
-	}
-
-	server.keeper.SetValidatorSetPreferences(ctx, msg.Delegator, setMsg)
+	server.keeper.SetValidatorSetPreferences(ctx, msg.Delegator, preferences)
```

Comment:
Im confused on why we call Set twice here

Reply:
these are 2 different functions `SetValidatorSetPreference` vs `SetValidatorSetPreference**s**` 

i'll rename it to something better 😅